src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -4318,6 +4318,8 @@ reset_daos_env_before_exec(void)
 		d_daos_inited     = false;
 		daos_debug_inited = false;
 		context_reset     = false;
+		/* all IO requests go through dfuse */
+		d_hook_enabled    = false;
```

Comment:
are there any open pool/container handles that need to be cleaned up here?

Reply:
Thank you!
Normally bash creates child process with fork(), then call exec() to start a new application shortly. It probably is rare to open a new DFS container between fork() and exec().
I can address this issue in a follow up PR. We need to save the current pid when daos_pool_connect() and daos_cont_open() are called. We can close pool/container handles before exec() if they are opened by current process.